paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/debezium/DebeziumRecordParser.java

```diff
@@ -105,13 +120,95 @@ private JsonNode getBefore(String op) {
     @Override
     protected void setRoot(String record) {
         JsonNode node = JsonSerdeUtil.fromJson(record, JsonNode.class);
+
+        hasSchema = false;
         if (node.has(FIELD_SCHEMA)) {
             root = node.get(FIELD_PAYLOAD);
+            JsonNode schema = node.get(FIELD_SCHEMA);
+            if (!isNull(schema)) {
+                parseSchema(schema);
+                hasSchema = true;
+            }
         } else {
             root = node;
         }
     }
 
+    private void parseSchema(JsonNode schema) {
+        debeziumTypes.clear();
+        classNames.clear();
+        parameters.clear();
+
+        ArrayNode schemaFields = getNodeAs(schema, "fields", ArrayNode.class);
+        Preconditions.checkNotNull(schemaFields);
+
+        ArrayNode fields = null;
+        for (int i = 0; i < schemaFields.size(); i++) {
+            JsonNode node = schemaFields.get(i);
+            if (getString(node, "field").equals("after")) {
+                fields = getNodeAs(node, "fields", ArrayNode.class);
+                break;
+            } else if (getString(node, "field").equals("before")) {
+                if (fields == null) {
+                    fields = getNodeAs(node, "fields", ArrayNode.class);
+                }
+            }
+        }
+        Preconditions.checkNotNull(fields);
+
+        for (JsonNode node : fields) {
+            String field = getString(node, "field");
+
+            debeziumTypes.put(field, getString(node, "type"));
+            classNames.put(field, getString(node, "name"));
+
+            String parametersString = getString(node, "parameters");
+            Map<String, String> parametersMap =
+                    parametersString == null
+                            ? Collections.emptyMap()
+                            : parseJsonMap(parametersString, String.class);
+            parameters.put(field, parametersMap);
+        }
+    }
+
+    @Nullable
+    private String getString(JsonNode node, String fieldName) {
+        JsonNode fieldValue = node.get(fieldName);
+        return isNull(fieldValue) ? null : fieldValue.asText();
+    }
+
+    @Override
+    protected Map<String, String> extractRowData(
+            JsonNode record, LinkedHashMap<String, DataType> paimonFieldTypes) {
+        if (!hasSchema) {
+            return super.extractRowData(record, paimonFieldTypes);
+        }
+
+        Map<String, String> recordMap =
+                JsonSerdeUtil.convertValue(record, new TypeReference<Map<String, String>>() {});
```

Comment:
There is a risk here. When the data is a complex type (such as array), json parsing will be abnormal.